webapp/src/client/client.ts

```diff
@@ -40,6 +41,41 @@ class SurveyClient extends HttpClient {
 
         return this.doPost(`${this.url}/survey/${surveyID}/end`);
     };
+
+    downloadSurveyReport = async (surveyID: string) => {
+        if (!surveyID || !ID_PATH_PATTERN.test(surveyID)) {
+            return Promise.reject(new Error('invalid survey ID encountered. Survey ID should be a 26 character, lowercase alphanumeric string'));
+        }
+
+        const url = `${this.url}/survey/${surveyID}/report`;
+
+        const response = await fetch(url, {
+            method: 'GET',
+            headers: {'Content-Type': 'application/zip'},
+        });
+
+        const blob = await response.blob();
+        const href = window.URL.createObjectURL(blob);
+        const link = document.createElement('a');
+        link.href = href;
+        link.setAttribute('download', this.extractFilename(response.headers.get('content-disposition'), 'survey_report'));
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+
+        return Promise.resolve();
+    };
+
+    extractFilename = (input: string | null, defaultName?: string): string => {
+        const regex = /filename\*?=["']?((?:\\.|[^"'\s])+)(?=["']?)/g;
```

Comment:
This might be worth checking with security for any regex matching issues.

Reply:
It's copied from MM webapp's `CommercialSupportModal` component, so nothing new. I've added a code comment to mention the source of this function for future reference.